### PR TITLE
Improve: Update dependencies / refactoring code

### PR DESCRIPTION
**Pull request by codsay**
- Changes
  - Update dependencies
  - Refactoring code for more consistency
  - Add more convenience methods / remove unnecessary arguments
  - Introduce uid (hashed of id) for WalletInfo
  - Improve wallet descriptor (icon, description) 

- [x] Tests pass
image

- [x] Appropriate changes to README are included in PR

### CODE REVIEW

File: src/user/core.ts [hunk 1]
```diff
@@ -117,16 +117,16 @@ export interface IUser {
...
 
   /**
    * Get wallet information
    * @param id
-   * @param legacyWallet
    */
-  getWalletInfo(id: string, legacyWallet?: boolean): WalletInfo;
+  getWalletInfo(wl: WalletInfo): WalletInfo;
+  getWalletInfo(id: WalletInfo): WalletInfo;
+  getWalletInfo(wlOrId: any): WalletInfo;
```
Comment (gaupoit) [inline src/user/core.ts]: Please have look this linter warnings or add comments to ignore them.

Comment (codsay) [reply src/user/core.ts]: These changes are removed

File: src/user/user.ts [hunk 2]
```diff
@@ -124,40 +124,83 @@ export class User implements IUser {
...
     let info: WalletInfo = null;
-    if (legacyWallet === true) {
-      if (this.legacyWallets) {
-        info = this.legacyWallets.filter((x) => x.key == id)[0];
+
+    // Start with HD wallet (as we should in this mode instead of legacy)
+    if (this.wallet) {
+      const derives = this.wallet.derivedWallets;
+      if (derives) {
+        for (let i = 0; i < derives.length; i++) {
```
Comment (djhahe) [inline src/user/user.ts]: should we you `this.legacyWallets.find`  instead of for loop ?

Comment (codsay) [reply src/user/user.ts]: oh nice, gonna use it

File: src/user/user.ts [hunk 3]
```diff
@@ -177,22 +220,35 @@ export class User implements IUser {
...
+   * @param {string} value - serialized user information
+   * @param {bool} encrypted - the value is encrypted
    */
-  public deserialize(value: string) {
-    const text = AESUtils.decrypt(this.password.getPassword(), value);
+  public deserialize(value: string, encrypted = true): void {
+    let text = value;
+    try {
+      if (encrypted) {
+        text = AESUtils.decrypt(this.password.getPassword(), value);
```
Comment (djhahe) [inline src/user/user.ts]: is it duplicated with line:238?

File: src/user/user.ts [hunk 4]
```diff
@@ -124,38 +124,67 @@ export class User implements IUser {
...
   public setWalletInfo(
     id: string,
-    info: string | WalletDescriptor,
-    legacyWallet?: boolean
+    info: string | WalletDescriptor
   ) {
-    let walletInfo = this.getWalletInfo(id, legacyWallet);
+    let walletInfo = this.getWalletInfo(id);
     if (!walletInfo) {
+      const isLegacyWallet = this.isLegacyWalletID(id);
```
Comment (djhahe) [inline src/user/user.ts]: ~~if we use uid here the value will alway be true, I think~~

Comment (codsay) [reply src/user/user.ts]: Yes, it is okay to true here, we're in inserting mode, we only accept derived HD path
either private key or uid will be failed in this state

File: src/user/user.ts [hunk 5]
```diff
@@ -124,38 +124,67 @@ export class User implements IUser {
...
+        let idx = derives.findIndex(x => x.id === id || x.uid === id);
+        if (idx >= 0) {
+          derives.splice(idx, 1);
+          return;
+        }
+      }
+    }
+
+    if (this.legacyWallets) {
+      for (let i = 0; i < this.legacyWallets.length; i++) {
```
Comment (djhahe) [inline src/user/user.ts]: for loop here should be removed

**Review by djhahe — APPROVED** [review]

**Pull request merged**